### PR TITLE
Add visits table

### PR DESCRIPTION
**Pull request by rcqx**
## Comments
* Create Vets table
* Add JOIN tables for species, specialisation and visits
* Inserted data as requested in vets, specialisation and visits tables
* Created custom queries for analytical research as requested by the project
* Screenshots of the process documented in day4 folders
* Update readme file

image
image
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -88,6 +88,54 @@ WHERE name IN ('Angemon', 'Boarmon');
...
 
+-- INSERT VETS DATA
+BEGIN;
+
+ALTER SEQUENCE vets_id_seq RESTART WITH 1;
+
+INSERT INTO vets(name, age, date_of_graduation) VALUES('William Tatcher', 45, '2000-04-23');
+INSERT INTO vets(name, age, date_of_graduation) VALUES('Maisy Smith', 26, '2019-01-17');
+INSERT INTO vets(name, age, date_of_graduation) VALUES('Stephanie Mendez', 64, '1981-05-04');
+INSERT INTO vets(name, age, date_of_graduation) VALUES('Jack Harkness', 38, '2008-06-06');
```
Comment (AnselemOdims) [inline data.sql:99]: - [ ] The requirement is for Jun 6. Please kindly adjust date so that it matches requirement

Comment (AnselemOdims) [inline data.sql:99]: - [ ] The requirement is for Jun 6. Please kindly adjust date so that it matches requirement

File: data.sql [hunk 2]
```diff
@@ -88,6 +88,54 @@ WHERE name IN ('Angemon', 'Boarmon');
...
+
+INSERT INTO specializations(vet_id, species_id) VALUES(1, 1);
+INSERT INTO specializations(vet_id, species_id) VALUES(3, 1);
+INSERT INTO specializations(vet_id, species_id) VALUES(3, 2);
+INSERT INTO specializations(vet_id, species_id) VALUES(4, 2);
+
+COMMIT;
+
+-- INSERT DATA IN VISITS TABLE
+ROLLBACK;
```
Comment (AnselemOdims) [inline data.sql]: - [x] Please avoid rolling back these changes as the instructions does not say you should

File: queries.sql [hunk 3]
```diff
@@ -118,4 +118,94 @@ GROUP BY o.full_name
...
+WHERE ve.name = 'William Tatcher'
+ORDER BY date_of_visit DESC
+LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT a.name, ve.name, v.date_of_visit
+FROM animals a 
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez';
```
Comment (AnselemOdims) [inline queries.sql]: - [x] This query only tells us the name of the animal, vet name and date of visit but the requirement is for how many **different** animals the vet saw. Please kindly adjust query to match requirement

File: queries.sql [hunk 4]
```diff
@@ -118,4 +118,94 @@ GROUP BY o.full_name
...
+-- How many visits were with a vet that did not specialize 
+-- in that animal's species?
+SELECT a.name as Pet_name, ve.name as Vet_Name, s.name AS vet_specialization
+FROM visits v
+JOIN vets ve ON v.vet_id = ve.id
+JOIN animals a ON v.animal_id = a.id
+JOIN specializations sp ON ve.id = sp.vet_id
+JOIN species s ON s.id = sp.species_id
+WHERE sp.species_id != a.species_id 
+AND ve.name != 'Stephanie Mendez';
```
Comment (AnselemOdims) [inline queries.sql]: - [x] The query says `**How many** times visits where with a vet that did not specialize in that animal's species?`. Please kindly adjust your query to output the number of times

Comment (AnselemOdims) [inline queries.sql]: - [x] The query says `**How many** times visits where with a vet that did not specialize in that animal's species?`. Please kindly adjust your query to output the number of times

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @jr-cast , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 1 above)
Comment (rcqx) [reply data.sql:99]: Hello @AnselemOdims could you be more specific? In line 99 I'm inserting June 6th, right?

Comment (rcqx) [reply data.sql:99]: Hello @AnselemOdims could you be more specific? In line 99 I'm inserting June 6th, right? Could you elaborate further? Thanks

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @jr-cast 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done making the required changes from the previous reviewer
- Great communication with the previous reviewer

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**